core/src/main/java/org/apache/iceberg/stats/StatsFormat.java

```diff
@@ -0,0 +1,141 @@
+  }
+
+  static int readIntegerLittleEndian(byte[] data, int offset) {
+    return Byte.toUnsignedInt(data[offset]) |
+        (Byte.toUnsignedInt(data[offset + 1]) << 8) |
+        (Byte.toUnsignedInt(data[offset + 2]) << 16) |
+        (Byte.toUnsignedInt(data[offset + 3]) << 24);
+  }
+
+  static ByteBuffer compressFooterPayload(ByteBuffer payload) {
+    return compress(StatsCompressionCodec.LZ4, payload);
+  }
+
+  static ByteBuffer decompressFooterPayload(ByteBuffer footer) {
+    return decompress(StatsCompressionCodec.LZ4, footer);
+  }
+
+  static ByteBuffer compressBlob(StatsCompressionCodec codec, ByteBuffer data) {
+    return compress(codec, data);
+  }
+
+  static ByteBuffer decompressBlob(StatsCompressionCodec codec, ByteBuffer data) {
+    return decompress(codec, data);
+  }
+
+  private static ByteBuffer compress(StatsCompressionCodec codec, ByteBuffer input) {
+    Compressor compressor = getCompressor(codec);
+    ByteBuffer output = ByteBuffer.allocate(compressor.maxCompressedLength(input.remaining()));
+    compressor.compress(input, output);
+    output.flip();
+    return output;
+  }
+
+  private static Compressor getCompressor(StatsCompressionCodec codec) {
+    switch (codec) {
+      case LZ4:
+        // TODO currently not supported
+        break;
+      case ZSTD:
+        return new ZstdCompressor();
+    }
+    throw new UnsupportedOperationException("Unsupported codec: " + codec);
+  }
+
+  private static ByteBuffer decompress(StatsCompressionCodec codec, ByteBuffer input) {
+    switch (codec) {
+      case LZ4:
+        // TODO requires LZ4 frame decompressor, e.g. https://github.com/airlift/aircompressor/pull/142
+        break;
+
+      case ZSTD: {
+        byte[] inputBytes;
+        int inputOffset;
+        int inputLength;
+        if (input.hasArray()) {
+          inputBytes = input.array();
+          inputOffset = input.arrayOffset();
+          inputLength = input.remaining();
+        } else {
+          // TODO implement ZstdDecompressor.getDecompressedSize for ByteBuffer to avoid copying
+          inputBytes = new byte[input.remaining()];
+          input.get(inputBytes);
+          inputOffset = 0;
+          inputLength = inputBytes.length;
+        }
+        byte[] decompressed =
+            new byte[Math.toIntExact(ZstdDecompressor.getDecompressedSize(inputBytes, inputOffset, inputLength))];
+        int decompressedLength =
+            new ZstdDecompressor().decompress(
+                inputBytes,
+                inputOffset,
+                inputLength,
+                decompressed,
+                0,
+                decompressed.length);
+        Preconditions.checkState(decompressedLength == decompressed.length, "Invalid decompressed length");
+        return ByteBuffer.wrap(decompressed);
+      }
+    }
+
+    throw new UnsupportedOperationException("Unsupported codec: " + codec);
+  }
+
+  private static byte[] swap(byte[] bytes) {
```

Comment:
swap is a bit unclear here since we actually are just reversing the byte order here. Maybe "invert" or "reverse"?

Reply:
renaming to `swapBytes`